Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java

```diff
@@ -141,37 +140,36 @@ public void onReceive(Context context, Intent intent) {
                     watchdogCallback.setDrone(drone);
                     scheduleWatchdog();
                     if (mAppPrefs.getWarningOnLostOrRestoredSignal()) {
-                        speak("Data link restored");
+                        speak(context.getString(R.string.speak_heartbeat_restored));
                     }
                     break;
 
                 case AttributeEvent.MISSION_ITEM_UPDATED:
                     int currentWaypoint = intent.getIntExtra(AttributeEventExtra.EXTRA_MISSION_CURRENT_WAYPOINT, 0);
-					if(currentWaypoint != 0) {
-						//Zeroth waypoint is the home location.
-						speak("Going for waypoint " + currentWaypoint);
-					}
+                    if (currentWaypoint != 0) {
+                        //Zeroth waypoint is the home location.
+                        speak(context.getString(R.string.speak_mission_item_updated) + currentWaypoint);
```

Comment:
@jandrop here you can use [string formatting](https://developer.android.com/reference/java/util/Formatter.html#format(java.lang.String, java.lang.Object...)) to include the current waypoint directly:
- `<string name="speak_mission_item_updated">"Going for waypoint %d"</string>`
- `speak(context.getString(R.string.speak_mission_item_updated, currentWaypoint))`